config/initializers/z_rails_5_compatibility.rb

```diff
@@ -12,10 +12,27 @@ def read_association(_preloader, record)
         record.public_send(association_name)
       end
     end
+
+    module PluginMigrations
+      def pending_migrations
+        if Rails.version.start_with?('5.2')
+          return true if Foreman.in_setup_db_rake?
+
+          @pending_migrations = ActiveRecord::MigrationContext.new(migrations_paths).needs_migration? if @pending_migrations.nil?
+
+          Rails.logger.debug("There are pending migrations. Please run foreman-rake db:migrate.") if @pending_migrations
+
+          @pending_migrations
+        else
+          super
+        end
+      end
+    end
   end
 end
 
 ActionView::Template::Handlers::Rabl.singleton_class.prepend Foreman::Rails5::RablTemplateHandlerExt
 if Rails.version.start_with?('5.2')
   CollectionLoader.prepend Foreman::Rails5::CollectionLoaderAssocRead
 end
+Foreman::Plugin.prepend Foreman::Rails5::PluginMigrations
```

Comment:
Given the override is only useful with 5.2, why don't you move this into the `Rails.version.start_with?('5.2')` above and keep it small?

Reply:
:dart: because I wasn't thinking while rebasing :) Thanks for noticing, looks better now :+1: